contracts/wrappers/QuestPartner.sol

```diff
@@ -0,0 +1,356 @@
+import "../oz/libraries/SafeERC20.sol";
+import "../utils/Owner.sol";
+import "../oz/utils/ReentrancyGuard.sol";
+import "../utils/Errors.sol";
+import "../QuestBoard.sol";
+import "../QuestTreasureChest.sol";
+import "../interfaces/ISimpleDistributor.sol";
+
+/** @title Warden Quest Creator contract */
+/// @author Paladin
+/*
+    Contract allowing to create (and manage) Quests
+    while sharing fees if the partner
+    Alows to set a balclist of claimers to be applied to all Quests
+    created by that contract
+*/
+
+contract QuestPartner is Owner, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    /** @notice 1e18 scale */
+    uint256 private constant UNIT = 1e18;
+    /** @notice Max BPS value (100%) */
+    uint256 private constant MAX_BPS = 10000;
+
+
+    QuestBoard public immutable board;
+
+    QuestTreasureChest public immutable chest;
+
+    address public immutable partner;
+
+    address public immutable feesReceiver;
+
+    uint256 public immutable partnerShare; //BPS
+
+    address[] public voterBlacklist;
+
+    uint256[] public partnerQuests;
+
+    mapping(uint256 => address) public creators;
+
+    mapping(uint256 => address) public rewardTokens;
+
+    bool public killed;
+
+
+    event NewPartnerQuest(uint256 indexed questID, address indexed creator);
+
+    event AddVoterBlacklist(address account);
+
+    event RemoveVoterBlacklist(address account);
+
+    event Killed();
+
+
+    modifier onlyPartner(){
+        if(msg.sender != partner) revert Errors.CallerNotAllowed();
+        _;
+    }
+
+    modifier onlyAllowed(){
+        if(msg.sender != partner && msg.sender != owner()) revert Errors.CallerNotAllowed();
+        _;
+    }
+
+    modifier notKilled(){
+        if(killed) revert Errors.Killed();
+        _;
+    }
+
+    constructor(
+        address _board,
+        address _chest,
+        address _partner,
+        address _receiver,
+        uint256 _share
+    ){
+        if(_board == address(0)) revert Errors.ZeroAddress();
+        if(_chest == address(0)) revert Errors.ZeroAddress();
+        if(_partner == address(0)) revert Errors.ZeroAddress();
+        if(_receiver == address(0)) revert Errors.ZeroAddress();
+        if(_share == 0) revert Errors.NullAmount();
+        if(_share >= 10000) revert Errors.InvalidParameter();
+
+        board = QuestBoard(_board);
+        chest = QuestTreasureChest(_chest);
+
+        partner = _partner;
+
+        feesReceiver = _receiver;
+
+        partnerShare = _share;
+    }
+
+    function getPartnerQuests() external view returns(uint256[] memory){
+        return partnerQuests;
+    }
+
+    function getBlacklistedVoters() external view returns(address[] memory){
+        return voterBlacklist;
+    }
+
+    function createQuest(
+        address gauge,
+        address rewardToken,
+        uint48 duration,
+        uint256 objective,
+        uint256 rewardPerVote,
+        uint256 totalRewardAmount,
+        uint256 feeAmount
```

Comment:
Internal math is handled by the QuestBoard contract.
It's done here before calling the board because it would require an external call before calculating the feeAmount.
So putting any random amount here would make the transaction revert when calling board.createQuest()